### PR TITLE
[RFC] Remove occurrences of mb_off2cells 

### PR DESCRIPTION
**Pull request by ZviRackover**
This is another step towards completion of #7401

### CODE REVIEW

File: src/nvim/screen.c [hunk 1]
```diff
@@ -4329,15 +4323,7 @@ static void screen_line(int row, int coloff, int endcol,
   unsigned max_off_to;
   int col = 0;
   int hl;
-  int force = FALSE;                    /* force update rest of the line */
-  int redraw_this                       /* bool: does character need redraw? */
-  ;
-  int redraw_next;                      /* redraw_this for next character */
-  int clear_next = FALSE;
-  int char_cells;                       /* 1: normal char */
-                                        /* 2: occupies two display cells */
```
Comment (justinmk) [inline src/nvim/screen.c]: In cases like this I don't think it's a clear gain to move the variable declarations, now the documentation is scattered.

Comment (ZviRackover) [reply src/nvim/screen.c]: Yes, I wasn't sure at what point we are losing the benefits of local variable scope to readability. Will try to roll back a bit.

Comment (ZviRackover) [reply src/nvim/screen.c]: Moved back most variable declarations in latest revision.

Comment (justinmk) [reply src/nvim/screen.c]: this still loses the documentation for `char_cells`. In general there's no need to move these declarations unless nothing is being lost.

Comment (ZviRackover) [reply src/nvim/screen.c]: Moved the declaration back to the top and restored the comments.

**Pull request merged**